activesupport/lib/active_support/isolated_execution_state.rb

```diff
@@ -4,53 +4,58 @@
 
 module ActiveSupport
   module IsolatedExecutionState # :nodoc:
-    @isolation_level = :thread
+    @isolation_level = nil
 
     Thread.attr_accessor :active_support_execution_state
     Fiber.attr_accessor :active_support_execution_state
 
     class << self
       attr_reader :isolation_level
+      attr_accessor :scope
```

Comment:
Given `isolation_level` is the public API, probably best to make this a reader and set it directly as `@scope` below, actually.

Reply:
`scope` needs to be public so that the various locking mechanisms in the connection pool code can access the current thread/fiber; also, the reaper code checks `Fiber/Thread#alive?` directly.

Reply:
Gotcha, should be resolved now